session/session.go

```diff
@@ -1048,6 +1048,7 @@ func (s *session) execute(ctx context.Context, sql string) (recordSets []sqlexec
 
 	// Step1: Compile query string to abstract syntax trees(ASTs).
 	startTS := time.Now()
+	s.GetSessionVars().StmtCtx.StartTime = startTS
```

Comment:
Don't forget the execute statement.

Reply:
The `execute` statement also enters this function. Is it already handled?

Reply:
No, I mean binary execute statement.

Reply:
Another question, if it is a multi-statements SQL, it will be only parsed once. How to deal with it properly?

Reply:
Actually, the binary protocol does not need parse, but I think it still needs to add the compile-time for it. We could leave the parse time to 0 for the binary protocol.

Reply:
> Actually, the binary protocol does not need parse, but I think it still needs to add the compile-time for it. We could leave the parse time to 0 for the binary protocol.

Agreed.

Reply:
> Another question, if it is a multi-statements SQL, it will be only parsed once. How to deal with it properly?

How about use a avg parse time? For example, suppose there are `N` statements to be parsed, and the total parse time is `durParse`, we use `durParse/N` as the parse time for each sql statement?

Reply:
> No, I mean binary execute statement.

Got it. Binary execute statement is handled now, PTAL.